### PR TITLE
themes: add dark terrastruct flagship theme

### LINKED ISSUE #1150: dark mode version of "flagship terrastruct" and "default" themes
For Terrastruct members, specs here:
https://www.figma.com/file/cEgPl2HYQIBv6zr5AwZ82h/%5BKW%5D-Themes?node-id=3733%3A131298&t=yBMWMcwMxhmCkp25-1
https://www.figma.com/file/cEgPl2HYQIBv6zr5AwZ82h/%5BKW%5D-Themes?node-id=3909%3A26993&t=yBMWMcwMxhmCkp25-1

For non-Terrastruct members, specs are attached as PDF:
[default dark mode coding.pdf](https://github.com/terrastruct/d2/files/11180641/default.dark.mode.coding.pdf)
[default dark mode.pdf](https://github.com/terrastruct/d2/files/11180642/default.dark.mode.pdf)
[flagship dark mode.pdf](https://github.com/terrastruct/d2/files/11180643/flagship.dark.mode.pdf)
[flagship terrastruct dark mode coding.pdf](https://github.com/terrastruct/d2/files/11180644/flagship.terrastruct.dark.mode.coding.pdf)

### PR DESCRIPTION
**Pull request by berniexie**
Closes #1150 

demo

### CODE REVIEW

**Review by alixander — CHANGES_REQUESTED** [review]
add an example in themes_test (https://github.com/alixander/d2/blob/master/e2etests/themes_test.go)

File: ci/release/changelogs/next.md [hunk 1]
```diff
@@ -12,3 +13,4 @@
 - Fixes grid layouts not applying on objects with a constant near [#1173](https://github.com/terrastruct/d2/issues/1173)
 - Fixes markdown header rendering in firefox and safari [#1177](https://github.com/terrastruct/d2/issues/1177)
 - Fixes a grid layout panic when there are fewer objects than rows [#1189](https://github.com/terrastruct/d2/issues/1189)
+- Fixes an issue where PNG exports that were too large were crashing [#1093](https://github.com/terrastruct/d2/issues/1093)
```
Comment (alixander) [inline ci/release/changelogs/next.md:16]: this is wrong pr

**Review by alixander — COMMENTED** [review]
Let's keep dark themes in the same band of IDs. https://github.com/terrastruct/d2/blob/cce35e5d051664fd4765062430c0ff5b26b105b9/d2themes/d2themes.go#L25

File: ci/release/changelogs/next.md (hunk 1 above)
Comment (berniexie) [reply ci/release/changelogs/next.md:16]: oh well, gotta include it now

File: ci/release/changelogs/next.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 - Export diagrams to `.pptx` (PowerPoint)[#1139](https://github.com/terrastruct/d2/pull/1139)
 - Customize gap size in grid diagrams with `grid-gap`, `vertical-gap`, or `horizontal-gap` [#1178](https://github.com/terrastruct/d2/issues/1178)
+- New dark theme "Dark Terrastruct Flagship" with the theme ID of `2` [#1150](https://github.com/terrastruct/d2/issues/1150)
```
Comment (alixander) [inline ci/release/changelogs/next.md:5]: id

**Review by alixander — CHANGES_REQUESTED** [review]

**Review by alixander — APPROVED** [review]

**Pull request merged**